packages/dcos-integration-test/extra/test_helpers.py

```diff
@@ -1,5 +1,133 @@
+import copy
 import json
+import uuid
 
+from dcos_test_utils import marathon
 
+TEST_APP_NAME_FMT = 'integration-test-{}'
+
+# make the expanded config available at import time to allow determining
+# which tests should run before the test suite kicks off
 with open('/opt/mesosphere/etc/expanded.config.json', 'r') as f:
     expanded_config = json.load(f)
+
+
+def marathon_test_app(
```

Comment:
imo function would be easier to read if you break it up at the top level conditional statements on the setting sections that are more than one or two lines. Something like

```python
def marathon_test_app(...):
    container_port = set_container_port(network)
    test_uuid = uuid.uuid4().hex
    app = ...
    app.update(set_healthchecks(...)
    app.update(set_container(...))
    app.update(set_networking(...))
    if host_constraint is not None:
        app['constraints'] = [['hostname', 'CLUSTER', host_constraint]]
   return app, test_uuid
```